src/provider/github.js

```diff
@@ -1,65 +1,89 @@
-import { createApolloFetch } from 'apollo-fetch';
+import {createApolloFetch} from 'apollo-fetch';
 
 const uri = 'https://api.github.com/graphql';
 
-const apolloFetch = createApolloFetch({ uri });
+const apolloFetch = createApolloFetch({uri});
 
-apolloFetch.use(({ options }, next) => {
+const parseIssues = (project) => (project && project.data && project.data.repository && project.data.repository.issues)
+  ? project.data.repository.issues
+  : { totalCount: 0 };
+
+apolloFetch.use(({
+  options
+}, next) => {
   if (!options.headers) {
     // eslint-disable-next-line no-param-reassign
     options.headers = {
-      authorization: `bearer ${process.env.REACT_APP_GITHUB_KEY}`,
+      authorization: `bearer ${process.env.REACT_APP_GITHUB_KEY}`
     };
   }
   next();
 });
 
-const totalIssuesGetRequest = project =>
-  new Promise((resolve) => {
-    apolloFetch({
-      query: `query IssueCount($owner: String!, $repositoryName: String!) {
+const totalIssuesGetRequest = project => new Promise((resolve) => {
+  apolloFetch({
```

Comment:
https://github.com/apollographql/apollo-fetch/blob/master/packages/apollo-fetch/src/apollo-fetch.ts#L121-L123

apolloFetch already returns a promise. nesting this in a Promise is confusing.

Reply:
It's because we use Promise.all 'Promise.all(github.getTotalIssueCountFor(repositories))'. When a repository has no issues the the promise rejects. So we don't want the Promise.all to fail if a single repo has no issues.

Reply:
I don't see how it justifies wrapping it with a Promise.
To prevent the failure of the Promise you can return a valid object from catch (see comment below)

Reply:
Removed the promise now thanks.